cli/src/main.rs

```diff
@@ -624,32 +624,23 @@ fn verification_key<T: FieldElement>(
         .with_setup_file(params.map(PathBuf::from))
         .with_backend(*backend_type);
 
-    let vkey = pipeline.verification_key()?;
-
-    write_verification_key_to_fs(vkey, dir);
+    let mut vkey_file = BufWriter::new(fs::File::create(dir.join("vkey.bin")).unwrap());
+    pipeline.write_verification_key(&mut vkey_file)?;
+    vkey_file.flush().unwrap();
+    log::info!("Wrote vkey.bin.");
```

Comment:
I find it odd that here the pipeline writes to FS, and in the setup case it returns the data and the user writes to FS.

Reply:
Because at the moment of writing that code the pipeline was the easiest way to collect all required artifacts at the same time. Try collecting the verification key in a different way and you'll see. You'll need to repeatedly call the Pipeline to get different artifacts, or call the specific crates directly.

Reply:
That said, I'm fine with this not being in Pipeline and if it's easy to remove it and do it like Setup is done, sounds good to me.